daemon/ct.go

```diff
@@ -65,21 +65,23 @@ func (d *Daemon) EnableConntrackGC() {
 		for {
 			sleepTime := time.Duration(GcInterval) * time.Second
 
-			d.endpointsMU.Lock()
+			d.endpointsMU.RLock()
 
 			for k := range d.endpoints {
 				e := d.endpoints[k]
+				e.Mutex.RLock()
 				if e.Consumable == nil {
+					e.Mutex.RUnlock()
 					continue
 				}
-
+				e.Mutex.RUnlock()
```

Comment:
Yes, that's true. So usually, mutexes are put before the variable they are protecting. That's something missing in the our mutexes. The endpoint ID is being written once, and that happens inside the endpoint with `NewEndpointFromChangeModel`, we don't need a mutex for the ID because of that. The `e.LogStatus` can be called because it's a public method and all public methods can be called since the lock will be locked by the receiver's function.

I'll move all mutex to the proper places and add a comment regarding which variables are they protecting.